### PR TITLE
added room placement

### PR DESCRIPTION
**Pull request by thejocky**
add rooms to be randomly placed in the map at a random size and a random amount

### CODE REVIEW

File: test_dungeon.py [hunk 1]
```diff
@@ -1,19 +1,31 @@
...
+
+#     expect = """┌────┬───┐
+# │    │   │
+# ├────┼───┤
+# │    │   │
+# └────┴───┘"""
+
+#     d = dungeon(10, 5)
+#     d.layout = layout
+#     assert d.print() == expect
```
Comment (dgbuckley) [inline test_dungeon.py]: Why did this get commented out?

File: test_dungeon.py [hunk 2]
```diff
@@ -1,19 +1,31 @@
...
+
+
+map = [[0 for j in range(10)] for k in range(10)]
+
+for k in range(2):
+    for j in range(2):
+        map[j+1][k+2] = 1
+
+for j in map:
+    print(j)
```
Comment (dgbuckley) [inline test_dungeon.py]: Is this a test? If so you should try to use the pytest operator `assert`.

File: main.py [hunk 3]
```diff
@@ -34,16 +34,37 @@ def __init__(self, x: int, y: int):
...
+        while fails < 5:
+            length = random.randint(14, 24)
+            width = random.randint(7, 12)
+            x = random.randrange(len(self.layout[0])-length)
+            y = random.randrange(len(self.layout)-width)
+            if self.generate_room(x, y, length, width):
+                fails = 0
+                print (1)
+            else:
+                fails += 1
```
Comment (dgbuckley) [inline main.py]: Do we need to keep this? It seems like we only need to generate the grid size then add rooms to that.

File: main.py [hunk 4]
```diff
@@ -61,7 +82,7 @@ def find_wall_type(self, x: int, y: int):
             if x != len(self.layout[0]) - 1:
                 if self.layout[y][x + 1] == tile.WALL:
                     value += 5
-        return walls[value]
+        return dungeon.walls[value]
```
Comment (dgbuckley) [inline main.py]: For standards sake this should be `self.walls`. Good catch though.

File: test_dungeon.py (hunk 1 above)
Comment (thejocky) [reply test_dungeon.py]: I didn't want to figure out how to get pytest

File: main.py (hunk 3 above)
Comment (thejocky) [reply main.py]: I don't see what u mean

Comment (dgbuckley) [reply main.py]: Nevermind, I see what you are doing. This make sense.

**Pull request merged**